packages/aws-cdk-lib/aws-appsync/lib/graphqlapi.ts

```diff
@@ -652,6 +682,8 @@ export class GraphqlApi extends GraphqlApiBase {
     this.arn = this.api.attrArn;
     this.graphqlUrl = this.api.attrGraphQlUrl;
     this.name = this.api.name;
+    this.graphQLEndpointArn = this.api.attrGraphQlEndpointArn;
+    this.visibility = props.visibility ?? Visibility.GLOBAL;
```

Comment:
we dont want to change the behavior of the construct. If we start setting this to a specific value when it has not been set, this will lead to a resource replacement. https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-appsync-graphqlapi.html#cfn-appsync-graphqlapi-visibility